src/macaron/dependency_analyzer/dependency_resolver.py

```diff
@@ -129,6 +130,17 @@ def add_latest_version(
         url_to_artifact: dict[str, set]
             Used to detect artifacts that have similar repos.
         """
+        if item["url"] == "" and item["version"] != "unspecified" and item["group"] and item["name"]:
+            gav = f"{item['group']}:{item['name']}:{item['version']}"
+            urls = find_repo(gav, ["scm.url"])
```

Comment:
Note that this method is performing a connection. This `add_latest_version` is called indirectly via this line in the test - https://github.com/oracle-samples/macaron/blob/1906f88c3d725ef01433301031b1bec0ea7829c6/tests/dependency_analyzer/cyclonedx/test_cyclonedx.py#L71
This is the call hierachy:
```
test_convert_components_to_artifacts -> convert_components_to_artifacts -> add_latest_version -> find_repo
```
Therefore, our unit tests still use network connection.

Reply:
Good point. Is this something we want to change?

Reply:
How about using a flag (which by default is on) to find repos  via contacting the remote server? That way we could control making network calls.